### PR TITLE
[Indexer] Add transactions by address indexing

### PR DESCRIPTION
**Pull request by bowenyang007**
### Description
It's nice to have a dedicated table to get all transactions that touch an address instead of a view. I'm adding this to the coin_processor because the coin processor already needs to process every transaction. We don't want this to be in default processor currently because we've had instances where default processor takes forever to run migrations and it stalls explorer and wallet.

What qualifies as a transaction associated with an address: 
1. account signed or is part of a multi sig / multi agent
2. account resources / events are directly modified
3. account owns an object where resources / events are modified (only 1 degree)

I called out a bunch of TODOs here. Notably what we're currently unable to do: 
* Recursively find owner of an object
* Find an owner of an object that was deleted
* Find owner of a table, recursively or not

P.S. This PR also fixes a bug where we're not actually recording object deletes. 

### Test Plan
image

### CODE REVIEW

File: crates/indexer/src/models/coin_models/account_transactions.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    /// a user account, an object, or a resource account.
+    /// We will consider all transactions that modify a resource or event associated with a particular account.
+    /// We will do 1 level of redirection for now (e.g. if it's an object, we will record the owner as account address).
+    /// We will also consider transactions that the account signed or is part of a multi sig / multi agent.
+    /// TODO: recursively find the parent account of an object
+    /// TODO: include table items in the detection path
+    pub fn from_transaction(
+        transaction: &Transaction,
+    ) -> anyhow::Result<HashMap<AccountTransactionPK, Self>> {
+        let (events, wscs, signatures, txn_version) = match transaction {
```
Comment (movekevin) [inline crates/indexer/src/models/coin_models/account_transactions.rs:41]: This looks like a fair amount of common code. Is there a common library to call to do this?

File: crates/indexer/src/models/v2_objects.rs [hunk 2]
```diff
@@ -120,35 +108,42 @@ impl Object {
     /// This should never really happen since it's very difficult to delete the entire resource group
```
Comment (movekevin) [inline crates/indexer/src/models/v2_objects.rs:108]: This is no longer true. Unnamed token objects can be burnt and deleted very thoroughly. For most use cases using aptos_token, we'll see clean deletions so let's make sure we handle this case well (can be in a separate PR though).

File: crates/indexer/src/models/v2_objects.rs [hunk 3]
```diff
@@ -157,4 +152,44 @@ impl Object {
             Ok(None)
         }
     }
+
+    /// This is actually not great because object owner can change. The best we can do now though
```
Comment (movekevin) [inline crates/indexer/src/models/v2_objects.rs:156]: What's the performance implication for doing this multiple times in the same processing logic? Is this cached somewhere? This can potentially lead to a lot more quota/capacity consumed when running processors on the client side against a grpc streaming service

File: crates/indexer/src/models/v2_objects.rs [hunk 4]
```diff
@@ -157,4 +152,44 @@ impl Object {
...
+    fn get_object_owner(
```
Comment (movekevin) [inline crates/indexer/src/models/v2_objects.rs:157]: I wonder if this can lead to some performance/failure issue for the processor if the data is stored in remote databases (e.g. in Hive). If a processor has to query the object owner multiple times, this means multiple remote calls

File: crates/indexer/src/models/v2_objects.rs [hunk 5]
```diff
@@ -157,4 +152,44 @@ impl Object {
...
+        conn: &mut PgPoolConnection,
+        object_address: &str,
+    ) -> anyhow::Result<CurrentObject> {
+        let mut retried = 0;
+        while retried < QUERY_RETRIES {
+            retried += 1;
+            match CurrentObjectQuery::get_by_address(object_address, conn) {
+                Ok(res) => {
+                    return Ok(CurrentObject {
```
Comment (movekevin) [inline crates/indexer/src/models/v2_objects.rs:166]: I thought we also planned to do this for table items?

Comment (bowenyang007) [reply crates/indexer/src/models/v2_objects.rs:166]: Did you read the summary lol. We can't do this for table items performantly.

File: crates/indexer/src/models/v2_objects.rs (hunk 4 above)
Comment (bowenyang007) [reply crates/indexer/src/models/v2_objects.rs:157]: We don't delete that often so it should be fine for now.

File: crates/indexer/src/models/v2_objects.rs (hunk 2 above)
Comment (bowenyang007) [reply crates/indexer/src/models/v2_objects.rs:108]: Yea I've seen this already so I'll change the comments.

File: crates/indexer/src/models/coin_models/account_transactions.rs (hunk 1 above)
Comment (bowenyang007) [reply crates/indexer/src/models/coin_models/account_transactions.rs:41]: This will need to be cleaned up but for now I don't have time to refactor all that code.

**Review by larry-aptos — APPROVED** [review]

File: crates/indexer/src/models/coin_models/account_transactions.rs [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    pub account_address: String,
+}
+
+impl AccountTransaction {
+    /// This table will record every transaction that touch an account which could be
+    /// a user account, an object, or a resource account.
+    /// We will consider all transactions that modify a resource or event associated with a particular account.
+    /// We will do 1 level of redirection for now (e.g. if it's an object, we will record the owner as account address).
+    /// We will also consider transactions that the account signed or is part of a multi sig / multi agent.
+    /// TODO: recursively find the parent account of an object
```
Comment (rtso) [inline crates/indexer/src/models/coin_models/account_transactions.rs:36]: Should this be done in post-processing instead of in the processor?

File: crates/indexer/src/models/v2_objects.rs [hunk 7]
```diff
@@ -117,38 +105,55 @@ impl Object {
...
+    /// using KV store
+    pub fn from_delete_resource(
         delete_resource: &DeleteResource,
         txn_version: i64,
         write_set_change_index: i64,
+        object_mapping: &HashMap<CurrentObjectPK, CurrentObject>,
+        conn: &mut PgPoolConnection,
     ) -> anyhow::Result<Option<(Self, CurrentObject)>> {
-        if delete_resource.resource.to_string() == "0x1::object::ObjectCore" {
+        if delete_resource.resource.to_string() == "0x1::object::ObjectGroup" {
```
Comment (rtso) [inline crates/indexer/src/models/v2_objects.rs:118]: What's object group and why do we need to change this?

File: crates/indexer/src/processors/default_processor.rs [hunk 8]
```diff
@@ -524,12 +526,54 @@ impl TransactionProcessor for DefaultTransactionProcessor {
         }
 
         // TODO, merge this loop with above
+        // Moving object handling here because we need a single object
```
Comment (rtso) [inline crates/indexer/src/processors/default_processor.rs:529]: What's the difference between this new way and the old way of getting the objects? They look like they're doing the same thing.

**Review by rtso — APPROVED** [review]

File: crates/indexer/src/models/coin_models/account_transactions.rs (hunk 6 above)
Comment (bowenyang007) [reply crates/indexer/src/models/coin_models/account_transactions.rs:36]: we might not have a choice lol.

File: crates/indexer/src/models/v2_objects.rs (hunk 7 above)
Comment (bowenyang007) [reply crates/indexer/src/models/v2_objects.rs:118]: the old thing wasn't working hahaha. I assuemd it would be ObjectCore but actually when it's deleted it's ObjectGroup. It was just buggy before.

File: crates/indexer/src/processors/default_processor.rs (hunk 8 above)
Comment (bowenyang007) [reply crates/indexer/src/processors/default_processor.rs:529]: They are. If you notice though that I'm passing all_objects into the from_delete_resource function. That's the main difference. I mean I could've made it work as before as well but the right thing to do is to use a general forloop to avoid the duplicate work so I just refactored this at the same time.

File: crates/indexer/src/models/v2_objects.rs (hunk 7 above)
Comment (rtso) [reply crates/indexer/src/models/v2_objects.rs:118]: that's weird that the resource type is different for writes and deletes.

**Review by movekevin — APPROVED** [review]
